nodebuilder/p2p/routing.go

```diff
@@ -6,70 +6,46 @@ import (
 
 	"github.com/ipfs/go-datastore"
 	dht "github.com/libp2p/go-libp2p-kad-dht"
-	"github.com/libp2p/go-libp2p/core/protocol"
 	"github.com/libp2p/go-libp2p/core/routing"
 	"go.uber.org/fx"
 
 	"github.com/celestiaorg/celestia-node/nodebuilder/node"
+	"github.com/celestiaorg/celestia-node/share/shwap/p2p/discovery"
 )
 
-// contentRouting constructs nil content routing,
-// as for our use-case existing ContentRouting mechanisms, e.g DHT, are unsuitable
-func contentRouting(r routing.PeerRouting) routing.ContentRouting {
-	return r.(*dht.IpfsDHT)
-}
-
-// peerRouting provides constructor for PeerRouting over DHT.
-// Basically, this provides a way to discover peer addresses by respecting public keys.
-func peerRouting(cfg *Config, tp node.Type, params routingParams) (routing.PeerRouting, error) {
-	opts := []dht.Option{
-		dht.BootstrapPeers(params.Peers...),
-		dht.ProtocolPrefix(protocol.ID(fmt.Sprintf("/celestia/%s", params.Net))),
-		dht.Datastore(params.DataStore),
-		dht.RoutingTableRefreshPeriod(cfg.RoutingTableRefreshPeriod),
-	}
-
-	if isBootstrapper() {
-		opts = append(opts,
-			dht.BootstrapPeers(), // no bootstrappers for a bootstrapper ¯\_(ツ)_/¯
-		)
-	}
-
```

Comment:
We need this 😑 , otherwise bootstrapping(OnStart hook) will deadlock bootstrappers